diff --git a/dist/index-BN0B0y8a.d.ts b/dist/index-BN0B0y8a.d.ts
index d8ca73f0c0165b6c6acfdd9fe72fbf4866a3007f..3ae0211d508c165a7f826cd93524d7a016a99a08 100644
--- a/dist/index-BN0B0y8a.d.ts
+++ b/dist/index-BN0B0y8a.d.ts
@@ -1195,21 +1195,21 @@ interface EXPERIMENTAL_ResolverRecord_Base {
   /**
    * Name of the matcher. Unique across all matchers. If missing, this record
    * cannot be matched. This is useful for grouping records.
    */
-  name?: RecordName;
+  name?: RecordName | undefined;
   /**
    * {@link MatcherPattern} for the path section of the URI.
    */
-  path?: MatcherPatternPath;
+  path?: MatcherPatternPath | undefined;
   /**
    * {@link MatcherPattern} for the query section of the URI.
    */
   query?: MatcherPatternQuery[];
   /**
    * {@link MatcherPattern} for the hash section of the URI.
    */
-  hash?: MatcherPatternHash;
+  hash?: MatcherPatternHash | undefined;
   /**
    * Parent record. The parent can be a group or a matchable record.
    * It will be included in the `matched` array of a resolved location.
    */
